sim/deathknight/anti_magic_shell.go

```diff
@@ -39,50 +39,88 @@ func (dk *Deathknight) registerAntiMagicShellSpell() {
 
+	physDmgTakenMult := dk.darkrunedPlateAMSBonus()
+	spellDmgTakenMult := 0.25
+
 	var targetDummySpell *core.Spell = nil
 	dk.AntiMagicShellAura = dk.RegisterAura(core.Aura{
 		Label:    "Anti-Magic Shell",
 		ActionID: actionID,
 		Duration: time.Second * 5,
 		OnGain: func(aura *core.Aura, sim *core.Simulation) {
-			target := aura.Unit.CurrentTarget
-			if targetDummySpell == nil && target != nil {
-				targetDummySpell = aura.Unit.CurrentTarget.RegisterSpell(core.SpellConfig{
-					ActionID:    core.ActionID{SpellID: 49375},
-					SpellSchool: core.SpellSchoolMagic,
+			if dk.Inputs.IsDps {
+				target := aura.Unit.CurrentTarget
+				if targetDummySpell == nil && target != nil {
+					targetDummySpell = aura.Unit.CurrentTarget.RegisterSpell(core.SpellConfig{
+						ActionID:    core.ActionID{SpellID: 49375},
+						SpellSchool: core.SpellSchoolMagic,
 
-					Cast: core.CastConfig{},
+						Cast: core.CastConfig{},
 
-					ApplyEffects: core.ApplyEffectFuncDirectDamage(core.SpellEffect{
-						ProcMask: core.ProcMaskSpellDamage,
+						ApplyEffects: core.ApplyEffectFuncDirectDamage(core.SpellEffect{
+							ProcMask: core.ProcMaskSpellDamage,
 
-						DamageMultiplier: 1,
+							DamageMultiplier: 1,
 
-						BaseDamage:     core.BaseDamageConfigRoll(dk.Inputs.AvgAMSHit*0.9, dk.Inputs.AvgAMSHit*1.1),
-						OutcomeApplier: target.OutcomeFuncAlwaysHit(),
-					}),
-				})
-			}
+							BaseDamage:     core.BaseDamageConfigRoll(dk.Inputs.AvgAMSHit*0.9, dk.Inputs.AvgAMSHit*1.1),
+							OutcomeApplier: target.OutcomeFuncAlwaysHit(),
+						}),
+					})
+				}
 
-			pa := &core.PendingAction{}
-			pa.NextActionAt = sim.CurrentTime + time.Duration(sim.RandomFloat("ams induced damage")*5.0*float64(time.Second))
-			pa.Priority = core.ActionPriorityAuto
-			pa.OnAction = func(sim *core.Simulation) {
-				if sim.RandomFloat("AMS trigger chance") < core.MinFloat(dk.Inputs.AvgAMSSuccessRate, 1.0) {
-					targetDummySpell.Cast(sim, aura.Unit)
+				pa := &core.PendingAction{}
+				pa.NextActionAt = sim.CurrentTime + time.Duration(sim.RandomFloat("ams induced damage")*5.0*float64(time.Second))
+				pa.Priority = core.ActionPriorityAuto
+				pa.OnAction = func(sim *core.Simulation) {
+					if sim.RandomFloat("AMS trigger chance") < core.MinFloat(dk.Inputs.AvgAMSSuccessRate, 1.0) {
+						targetDummySpell.Cast(sim, aura.Unit)
+					}
 				}
+				sim.AddPendingAction(pa)
 			}
-			sim.AddPendingAction(pa)
+
+			dk.PseudoStats.PhysicalDamageTakenMultiplier *= physDmgTakenMult
+			dk.PseudoStats.ArcaneDamageTakenMultiplier *= spellDmgTakenMult
+			dk.PseudoStats.FireDamageTakenMultiplier *= spellDmgTakenMult
+			dk.PseudoStats.FrostDamageTakenMultiplier *= spellDmgTakenMult
+			dk.PseudoStats.HolyDamageTakenMultiplier *= spellDmgTakenMult
+			dk.PseudoStats.NatureDamageTakenMultiplier *= spellDmgTakenMult
+			dk.PseudoStats.ShadowDamageTakenMultiplier *= spellDmgTakenMult
+			dk.PseudoStats.DiseaseDamageTakenMultiplier *= spellDmgTakenMult
```

Comment:
This might be double-counting, aren't diseases also spells?

Reply:
Probably, I've removed it for now as that makes sense, when we test the tank stuff I'll verify it